File: WWDC2015/WWDC2015/NBMapViewAnnotation.m
#import "NBMapViewAnnotation.h"

@interface NBMapViewAnnotation ()

@property (nonatomic) NSString *title;
@property (nonatomic) CLLocationCoordinate2D coordinate;

@end

@implementation NBMapViewAnnotation

- (id)initWithCoordinate:(CLLocationCoordinate2D)coordinate name:(NSString *)name {
    self = [super init];
    if (self) {
        self.coordinate = coordinate;
        self.title = name;
    }
    return self;
}

@end

File: WWDC2015/NBMapViewAnnotation.swift
import MapKit
import Foundation

@objc class NBMapViewAnnotation : NSObject, MKAnnotation {
    var title: String
    var coordinate: CLLocationCoordinate2D
    
    init(coordinate: CLLocationCoordinate2D, name:NSString) {
        self.title = name as String;
        self.coordinate = coordinate;
    }
}